pkg/services/ngalert/state/state.go

```diff
@@ -400,19 +395,38 @@ func resultKeepLast(state *State, rule *models.AlertRule, result eval.Result, lo
 	}
 }
 
-func (a *State) NeedsSending(resendDelay time.Duration) bool {
-	switch a.State {
-	case eval.Pending:
-		// We do not send notifications for pending states
+// NeedsSending returns true if the given state needs to be sent to the Alertmanager.
+// Reasons for sending include:
+// - The state has been resolved since the last notification.
+// - The state is firing and the last notification was sent at least resendDelay ago.
+// - The state was resolved within the resolvedRetention period, and the last notification was sent at least resendDelay ago.
+func (a *State) NeedsSending(resendDelay time.Duration, resolvedRetention time.Duration) bool {
```

Comment:
NOTE: Decent amount of logic changes here, but at the very least it brings us closer to the logic in prometheus ruler.

Reply:
Yes, but in my opinion it is harder to read now.